drizzle-orm/src/mysql-core/query-builders/select.ts

```diff
@@ -260,6 +268,61 @@ export abstract class MySqlSelectQueryBuilderBase<
 
 	fullJoin = this.createJoin('full');
 
+	private createSetOperator(
+		type: SetOperator,
+		isAll: boolean,
+	): <TValue extends MySqlSetOperatorWithResult<TResult>>(
+		rightSelection:
+			| ((setOperators: GetMySqlSetOperators) => SetOperatorRightSelect<TValue, TResult>)
+			| SetOperatorRightSelect<TValue, TResult>,
+	) => MySqlSelectWithout<
+		this,
+		TDynamic,
+		MySqlSetOperatorExcludedMethods,
+		true
+	> {
+		return (rightSelection) => {
+			const rightSelect = (typeof rightSelection === 'function'
+				? rightSelection(getMySqlSetOperators())
+				: rightSelection) as TypedQueryBuilder<
+					any,
+					TResult
+				>;
+
+			if (!haveSameKeys(this.getSelectedFields(), rightSelect.getSelectedFields())) {
+				throw new Error(
```

Comment:
This is the code that throws if different `SelectedField` keys are detected. We might want to think if we want to keep this code. @dankochetov

Reply:
Let's leave it like this for now, it's better to gather feedback first.